sw/device/lib/crypto/impl/ecc.c

```diff
@@ -12,25 +14,48 @@
 crypto_status_t otcrypto_ecdsa_keygen(const ecc_curve_t *elliptic_curve,
                                       crypto_blinded_key_t *private_key,
                                       ecc_public_key_t *public_key) {
-  // TODO: Connect ECDSA operations to API.
-  return kCryptoStatusNotImplemented;
+  crypto_status_t err =
+      otcrypto_ecdsa_keygen_async_start(elliptic_curve, private_key->config);
+  // TODO(#17803): replace this error check with HARDENED_TRY if cryptolib and
+  // status_t errors become more compatible.
+  if (launder32(err) != kCryptoStatusOK) {
+    return err;
+  }
+  HARDENED_CHECK_EQ(err, kCryptoStatusOK);
+  return otcrypto_ecdsa_keygen_async_finalize(elliptic_curve, private_key,
+                                              public_key);
 }
 
 crypto_status_t otcrypto_ecdsa_sign(const crypto_blinded_key_t *private_key,
                                     crypto_const_uint8_buf_t input_message,
                                     const ecc_curve_t *elliptic_curve,
                                     ecc_signature_t *signature) {
-  // TODO: Connect ECDSA operations to API.
-  return kCryptoStatusNotImplemented;
+  crypto_status_t err = otcrypto_ecdsa_sign_async_start(
+      private_key, input_message, elliptic_curve);
+  // TODO(#17803): replace this error check with HARDENED_TRY if cryptolib and
+  // status_t errors become more compatible.
+  if (launder32(err) != kCryptoStatusOK) {
+    return err;
+  }
+  HARDENED_CHECK_EQ(err, kCryptoStatusOK);
+  return otcrypto_ecdsa_sign_async_finalize(elliptic_curve, signature);
```

Comment:
nit: TODO or feel free to ignore if planned for later or handled in a different way: Should we run `edca_verify` before returning to the caller? (defense in depth)